mart/attack/enforcer.py

```diff
@@ -95,36 +97,32 @@ def verify(self, input_adv, *, input, target):
 
 
 class Enforcer:
-    def __init__(self, constraints: dict[str, Constraint]) -> None:
-        self.constraints = list(constraints.values())  # intentionally ignore keys
+    def __init__(self, **modality_constraints: dict[str, dict[str, Constraint]]) -> None:
+        self.modality_constraints = modality_constraints
 
     @torch.no_grad()
     def __call__(
         self,
-        input_adv: torch.Tensor | Iterable[torch.Tensor],
+        input_adv: torch.Tensor | Iterable[torch.Tensor] | Iterable[dict[str, torch.Tensor]],
         *,
-        input: torch.Tensor | Iterable[torch.Tensor],
-        target: torch.Tensor | Iterable[torch.Tensor | dict[str, Any]],
+        input: torch.Tensor | Iterable[torch.Tensor] | Iterable[dict[str, torch.Tensor]],
+        target: torch.Tensor | Iterable[torch.Tensor] | Iterable[dict[str, Any]],
```

Comment:
Can you be more specific than `Any` with the types?

Reply:
Fixed. I think it can be a tensor or a string (file name for example).

Reply:
I changed it back to Any because it doesn't really matter. Other modules that use modality_dispatch() also annotate that as Any.